### PR TITLE
kv: resolve three compilation warnings

### PR DESCRIPTION
**Pull request by EdwardVitor**
Resolve three warning issues in the derived classes overriding base class virtual functions with mismatched parameter lists at lines 69, 72, and 74 in src/kv/rocksdb_cache/ShardedCache.h.

```
with ../src/kv/rocksdb_cache/ShardedCache.h:74:16: warning:   by 'virtual bool rocksdb_cache::ShardedCache::Release(rocksdb::Cache::Handle*, bool)' [-Woverloaded-virtual]
   virtual bool Release(rocksdb::Cache::Handle* handle, bool force_erase = false) override;
                ^~~~~~~
../src/kv/rocksdb_cache/ShardedCache.h:72:35: warning:   by 'virtual rocksdb::Cache::Handle* rocksdb_cache::ShardedCache::Lookup(const rocksdb::Slice&, rocksdb::Statistics*)' [-Woverloaded-virtual]
   virtual rocksdb::Cache::Handle* Lookup(const rocksdb::Slice& key, rocksdb::Statistics* stats) override;
                                   ^~~~~~
../src/kv/rocksdb_cache/ShardedCache.h:69:27: warning:   by 'virtual rocksdb::Status rocksdb_cache::ShardedCache::Insert(const rocksdb::Slice&, void*, size_t, rocksdb::Cache::DeleterFn, rocksdb::Cache::Handle**, rocksdb::Cache::Priority)' [-Woverloaded-virtual]
   virtual rocksdb::Status Insert(const rocksdb::Slice& key, void* value, size_t charge,
                           ^~~~~~
```

Signed-off-by: cuiming <cuiming_yewu@cmss.chinamobile.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
@EdwardVitor please remove "cmake" from the title of the commit message. it has nothing to do with CMake.

**Comment by EdwardVitor** [comment]
> @EdwardVitor please remove "cmake" from the title of the commit message. it has nothing to do with CMake.

Ok, thanks

**Comment by EdwardVitor** [comment]
@tchaikov @cbodley

**Comment by EdwardVitor** [comment]
@tchaikov @cbodley @anthonyeleven @idryomov @batrick please help review

**Comment by anthonyeleven** [comment]
No docs content here, out of my demesne

File: src/kv/rocksdb_cache/ShardedCache.h [hunk 1]
```diff
@@ -62,6 +62,9 @@ class CacheShard {
 // Keys are sharded by the highest num_shard_bits bits of hash value.
 class ShardedCache : public rocksdb::Cache, public PriorityCache::PriCache {
  public:
+  using rocksdb::Cache::Release;
```
Comment (rzarzynski) [inline src/kv/rocksdb_cache/ShardedCache.h:65]: nit: how about putting these `usings` just above corresponding method declarations? I'm thinking about some like:

```cpp
  using rocksdb::Cache::Release;
  virtual bool Release(rocksdb::Cache::Handle* handle, bool force_erase = false) override;
```

**Review by rzarzynski — APPROVED** [review]
nit: the commit title should follow the convention -- there should an extra space just after the colon:

```
 kv: resolve three complication warnings 
```

Anyway, LGTM. The comments are about minors, they are not blockers.

File: src/kv/rocksdb_cache/ShardedCache.h (hunk 1 above)
Comment (EdwardVitor) [reply src/kv/rocksdb_cache/ShardedCache.h:65]: Great suggestion, thank u~

**Comment by EdwardVitor** [comment]
@markhpc @aclamk thank u for review

**Comment by EdwardVitor** [comment]
@markhpc @cbodley Can you please help me take a look if it can be approved?

**Comment by zdover23** [comment]
@EdwardVitor: I'm a docs guy, so i wouldn't dare to touch a header file. I'll ping Casey in a few hours and bring this to his attention, though.

**Comment by cbodley** [comment]
> @markhpc @cbodley @zdover23 Can you please help me take a look if it can be approved?

this has been approved and tagged 'needs-qa'. it can merge once someone runs it through teuthology testing

**Comment by EdwardVitor** [comment]
> @EdwardVitor: I'm a docs guy, so i wouldn't dare to touch a header file. I'll ping Casey in a few hours and bring this to his attention, though.

ok,thanks

**Comment by EdwardVitor** [comment]
> > @markhpc @cbodley @zdover23 Can you please help me take a look if it can be approved?
> 
> this has been approved and tagged 'needs-qa'. it can merge once someone runs it through teuthology testing

soga, thank u

**Comment by ljflores** [comment]
Rados approved: https://tracker.ceph.com/projects/rados/wiki/MAIN#httpstrellocomck9RNURve1897-wip-yuri3-testing-2023-12-07-0727-old-wip-yuri3-testing-wip-neorados-learning-from-experience

**Pull request merged**